docs/reference/search/search-your-data/knn-search.asciidoc

```diff
@@ -0,0 +1,249 @@
+[[knn-search]]
+== k-nearest neighbor (kNN) search
+++++
+<titleabbrev>kNN search</titleabbrev>
+++++
+
+//tag::knn-def[]
+A _k-nearest neighbor_ (kNN) search finds the _k_ nearest vectors to a query
+vector, as measured by a similarity metric.
+//end::knn-def[]
+
+image::images/kNN/knn_diagram.svg[Approximate kNN search as a two-dimensional graph,align="center"]
+
+Common use cases for kNN include:
+
+* Relevance ranking based on natural language processing (NLP) algorithms
+* Product recommendations and recommendation engines
+* Similarity search for images or videos
+* Outlier detection and classification
+
+[discrete]
+[[knn-prereqs]]
+=== Prerequisites
+
+* To run a kNN search, you must be able to convert your data into meaningful
```

Comment:
Thanks for the suggestion. I updated this to:

> To run a kNN search, you must be able to convert your data into meaningful vector values. You create these vectors outside of Elasticsearch and add them to documents as [`dense_vector`](http://localhost:8000/guide/dense-vector.html) field values. Queries are represented as vectors with the same dimension.
>
> Design your vectors so that the closer a document’s vector is to a query vector, based on a similarity metric, the better its match.